datafusion/src/physical_plan/analyze.rs

```diff
@@ -82,6 +83,10 @@ impl ExecutionPlan for AnalyzeExec {
         Partitioning::UnknownPartitioning(1)
     }
 
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
```

Comment:
Having to sprinkle `output_ordering` around was annoying -- but I think it may be worth it to try and avoid some nasty bugs.

Reply:
Agree, makes sense to be explicit